lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4155,6 +4215,13 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
       auto addActiveValue = [&](SILValue v) {
         if (visited.count(v))
           return;
+        // Diagnose active enum values. Differentiation of enum values is not
```

Comment:
Diagnosing active enum values is necessary because adjoint generation doesn't propagate adjoint values of enum associated values correctly.

Support is non-trivial because `switch_enum` operand and successor block arguments have different types: the operand has an enum type but successor block arguments have associated values' type. Adjoint value propagation needs to construct enum adjoint value from associated values' adjoint values.